Makefile

```diff
@@ -308,7 +308,7 @@ create-management-cluster: $(KUSTOMIZE) $(ENVSUBST) $(KUBECTL) $(KIND) ## Create
 
 	# Deploy CAPZ
 	$(KIND) load docker-image $(CONTROLLER_IMG)-$(ARCH):$(TAG) --name=$(KIND_CLUSTER_NAME)
-	$(KUSTOMIZE) build config/default | $(ENVSUBST) | $(KUBECTL) apply -f -
+	$(KUSTOMIZE) build config/default | $(ENVSUBST) | $(KUBECTL) apply -f - --server-side=true
```

Comment:
What does this change accomplish? Mostly for my own understanding

Reply:
How does this affect installation with clusterctl init or other mechanisms?

Reply:
clusterctl init doesn't do exactly the same thing as `kubectl apply` AFAIK where this would come into play. I'll try this again manually to make sure this isn't an issue there, but we would be seeing a similar error in e2e if it was, right?

Reply:
Just confirmed that `clusterctl init` completes successfully and all the CRDs get installed.

For any particular method of installing CAPZ, I'm not sure there's anything we can can realistically do here to prevent this specific issue since it essentially boils down to "resource too big" and we can't control how big the ASO CRDs are. If this could be proactively prevented, it would probably make the most sense to make that kind of change in ASO since it affects anyone installing the ManagedCluster CRD.